app/controllers/application_modal_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ApplicationModalController < ApplicationController
```

Comment:
Currently I'm thinking of a better approach which will be flexible and easy to use. Actually I don't really like "prepare_to_respond" and overriden methods. Yet at the same time this solution may be pretty flexible.